### PR TITLE
Use the 'confirm_advance' template to allow refreshing of totals when the order is not in the confirm state.

### PR DESCRIPTION
**Pull request by athal7**

### CODE REVIEW

**Comment by jhawthorn** [comment]
Is it possible to get a failing feature spec to go with this? The controller spec tests the new behaviour, but I don't know why that is needed or what this does.

**Comment by athal7** [comment]
@jhawthorn it's a bit tricky because this is kind of a catch-all for orders that are supposed to be in the confirm state (i.e. have payment, items, and address information), but for some reason (state machine?) have not quite made it into the confirm state.

File: backend/spec/controllers/spree/admin/orders_controller_spec.rb [hunk 1]
```diff
@@ -202,17 +202,28 @@
...
+          expect(response.status).to eq 200
+          expect(response).to render_template(:confirm)
+        end
+      end
 
-        it 'is successful' do
+      context 'when before confirm' do
+        before { order.stub completed?: false, confirm?: false }
+
+        it 'it renders the confirm_advance template (to allow refreshing of the order)' do
```
Comment (magnusvk) [inline backend/spec/controllers/spree/admin/orders_controller_spec.rb]: nitpick on the double it

Comment (athal7) [inline backend/spec/controllers/spree/admin/orders_controller_spec.rb]: whoops ya

File: backend/spec/controllers/spree/admin/orders_controller_spec.rb [hunk 2]
```diff
@@ -202,17 +202,28 @@
         spree_get :confirm, id: order.number
       end
 
-      context 'when incomplete' do
-        before { allow(order).to receive_messages :completed? => false }
+      context 'when in confirm' do
+        before { allow(order).to receive_messages completed?: false, confirm?: true }
+
+        it 'is renders the confirm page' do
```
Comment (magnusvk) [inline backend/spec/controllers/spree/admin/orders_controller_spec.rb]: and the is here seems superfluous also. adding value with my nitpicks every day!

**Comment by magnusvk** [comment]
:+1:

**Comment by magnusvk** [comment]
A question, though: This is the refresh totals screen, right? I've always wondered why we don't just go ahead and do that automatically and save the ninja an extra click. But happy to address that idea separately.

**Comment by athal7** [comment]
@jordan-brough might have more information about this situation actually

**Comment by jordan-brough** [comment]
I don't think it's a good idea to trigger mutations (like `advance`) on GET requests.  Maybe that's not the pragmatic thing to do given how our state machine stuff is set up, but that was the idea.

**Comment by athal7** [comment]
@jordan-brough that makes sense. can you help us understand more about how orders get into this state? is it when denormalized calculations don't add up to other totals?

**Comment by magnusvk** [comment]
@jordan-brough :+1: on not triggering mutations in GET requests, I didn't think of that.

**Comment by jordan-brough** [comment]
@athal7 I'm not sure what the most common scenario is, but if an admin picked up an in-progress order the order might not be in the "confirm" state already though it could be ready to be moved to the confirm state.  And there's no obvious way in the admin to trigger "order.advance", so when you are in such a scenario this might be the most obvious way to advance the order?  I'm sure there are better ways we could handle situations like that from a UI perspecitve.

**Comment by athal7** [comment]
:+1: thanks for the explanation

**Pull request merged**